src/Compiler/CompilerOptions.cpp

```diff
@@ -208,6 +208,11 @@ llvm::cl::opt<bool> enableSimdDataLayout("simd-data-layout",
                    "Set to 'true' if you want to enable SIMD optimizations."),
     llvm::cl::init(false), llvm::cl::cat(OnnxMlirOptions));
 
+llvm::cl::opt<bool> enableONNXHybridPass("onnx-hybrid-pass",
+    llvm::cl::desc("Enable ONNX hybrid pass (default=true)\n"
+                   "Set to 'false' if you want to disable ONNX hybrid pass."),
+    llvm::cl::init(true), llvm::cl::cat(OnnxMlirCommonOptions));
```

Comment:
I temporarily changed the default to `true` to surface the lit test failures that results from changing shape inference to the new algorithm

I also fixed half the lit test failures

Reply:
I figured out the linux CI build failures, it was another crash on an unranked input, this time when compiling the mnist-8 model from the model zoo. The crash was in FuseAddConvNullBiasPattern in src/Dialect/ONNX/Rewrite.td. I added a check `(HasShapeAndRank:$res)` to bypass the pattern when the type of ONNXAddOp is unranked.

Reply:
On second thought the better fix is to apply to shape inference patterns before the canonicalization patterns in the hybrid pass. In this way the shapes will be inferred before FuseAddConvNullBiasPattern is tested and the pattern will apply. I made this change and moved the added `(HasShapeAndRank:$res)` check to a standalone PR #2159.

Reply:
> I temporarily changed the default to true to surface the lit test failures that results from changing shape inference to the new algorithm

I changed the default back to false.